pkg/controllers/status/work_status_controller.go

```diff
@@ -312,23 +312,28 @@ func (c *WorkStatusController) reflectStatus(work *workv1alpha1.Work, clusterObj
 	}
 	c.EventRecorder.Eventf(work, corev1.EventTypeNormal, events.EventReasonReflectStatusSucceed, "Reflect status for object(%s/%s/%s) succeed.", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName())
 
-	if statusRaw == nil {
-		return nil
-	}
-
 	var resourceHealth workv1alpha1.ResourceHealth
-	// When an unregistered resource kind is requested with the ResourceInterpreter,
-	// the interpreter will return an error, we treat its health status as Unknown.
-	healthy, err := c.ResourceInterpreter.InterpretHealth(clusterObj)
-	if err != nil {
-		resourceHealth = workv1alpha1.ResourceUnknown
-		c.EventRecorder.Eventf(work, corev1.EventTypeWarning, events.EventReasonInterpretHealthFailed, "Interpret health of object(%s/%s/%s) failed, err: %s.", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), err.Error())
-	} else if healthy {
+	if statusRaw == nil {
+		// When a resource didn't implement ReflectStatus and its status field didn't collect from member cluster,
```

Comment:
> When a resource didn't implement ReflectStatus and its status field didn't collect from member cluster,

We have corresponding implementations for all resources. `reflectWholeStatus`

Reply:
I don't quite get it.
Do you mean you also take `empty` status as healthy? But implementing it in resource interpreter?

Reply:
I get it, you mean that all resource status will be collected by `reflectWholeStatus`:
https://github.com/karmada-io/karmada/blob/88bed31b92e1de6297be845c2284d96482f14e0e/pkg/resourceinterpreter/default/native/default.go#L151

Reply:
hi @chaunceyjiang, two questions:

1. how you handle resource like `clusterip` type `Service`, it exist in `e.reflectStatusHandlers[object.GroupVersionKind()]` but will return nil status in `reflectServiceStatus` now.
2. you mean you expect not return nil status when reflect no status filed resources, then may be should also have a default `InterpretHealth` function for this resources?

Reply:
changed to `continue to execute InterpretHealth even if statusRaw is nil`